src/Hamiltonians/reduced_density_matrix.jl

```diff
@@ -230,14 +230,16 @@ end
 function (calc!::ReducedDensityMatrixCalculcator!{TT, P})(result, pair) where {TT, P}
     addr_right, val_right = pair
     left = calc!.left
-
-    for j in axes(result, 2)
+    dim = calc!.dim
+    
+    for j in 1:dim
         dsts = find_mode(addr_right, vertices(j, Val(P)))
-        for i in axes(result, 1)
+        for i in j:dim
             srcs = reverse(find_mode(addr_right, vertices(i, Val(P))))
 
             addr_left, elem = excitation(addr_right, dsts, srcs)
             @inbounds result[i, j] += TT(conj(left[addr_left]) * elem * val_right)
+            @inbounds result[j, i] = result[i, j]
```

Comment:
Don't you need a complex conjugation here to make sure that the reduced density matrix is Hermitian?

Reply:
Also, isn't this line over writing some previously written results?

Reply:
In the last version, the operator calculated all elements of ``ReducedDensityMatrix`` (RDM); however, here it's only calculating the diagonal elements and upper triangular elements. This is by imposing the Hermitian property of RDM. This is making the code 40-50 % faster.

Reply:
Ah, I now see that I misread the code when reading it very quickly yesterday. Indeed, the lower triangular part is omitted from the for loops.

For this to work correctly with complex valued wave functions, I still think that you need a complex conjugation here, no?
```suggestion
            @inbounds result[j, i] = conj(result[i, j])
```

The other issue that I see is that you are not actually sampling the lower triangular matrix elements. While I understand that it is faster, you are also ignoring some information. I think that the stochastic noise in the lower triangle should be statistically independent from the upper one. Calculating both and taking the average should mitigate some of the fluctuations. I thus expect that you will get noisier results with this change. It may be worth testing what is better, i.e. how much data do you have to collect in order to get the same error bars?

Reply:
I understand that the stochastic noise will be different for the lower triangular element compared to the upper one, but it will not affect the result. We only need to focus on calculating the upper triangular elements, with the corresponding lower triangular elements set to the same value.